src/bin/bat/app.rs

```diff
@@ -223,6 +223,12 @@ impl App {
                         .unwrap_or_default(),
                 ),
             },
+            offset: self.matches.value_of("offset").map_or(0, |v|
+                        v.parse::<usize>().unwrap()
+                        + if style_components.header() {1} else {0} // the header is 1 row
```

Comment:
I think it would be better if we add a method to the style components that returns their potential header-height in lines. This way, we could use something like:
```
style_components.map(|s| s.height_in_header()).sum()
```
to get the total height.